File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/rsp/RspRetailerAccountList.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.rsp

import org.json.JSONArray
import org.json.JSONObject

class RspRetailerAccountList(
    private val accounts: MutableList<RspRetailerAccount>) : Rsp {

    override fun toJson(): JSONObject =
        JSONObject()
            .put("accounts", JSONArray(accounts.map { account -> account.toJson() }))
}

File: ios/Src/Mytiki/Sdk/Capture/Receipt/Capacitor/Rsp/RspRetailerAccountList.swift
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import BlinkReceipt
import BlinkEReceipt
import Capacitor

struct RspRetailerAccountList : Rsp{
    private let accounts: Array<RspRetailerAccount>
    
    func toJson() -> JSObject {
        JSObject.updateValue("accounts", JSONArray(accounts.map { account -> account.toJson() }))

    }
}
